lib/fastlane/actions/reset_simulator.rb

```diff
@@ -0,0 +1,65 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      RESET_SIMULATOR_CUSTOM_VALUE = :RESET_SIMULATOR_CUSTOM_VALUE
+    end
+
+    class ResetSimulatorAction < Action
+      def self.run(params)
+        require 'json'
+
+        unless Actions.sh_no_action("xcrun simctl list > /dev/null 2>&1 && echo 1 ||  echo 0", log: false).chomp.to_i == 1
+          Helper.log.error "Xcode version 7 or higher is required!"
+          return
+        end
+
+        json = JSON.parse(Actions.sh_no_action("xcrun simctl list --json devices", log: false))
+        json["devices"].each do |os, devices|
+          devices.each do |device|
+            next unless params[:device] == "all" ||
+              params[:device] == device["udid"] ||
+              params[:device] == device["name"] && (params[:os].nil? || params[:os] == os)
+
+            Actions.sh("xcrun simctl shutdown #{device['udid']}") if device["state"] == "Booted"
+            Actions.sh("xcrun simctl erase #{device['udid']}")
+
+            # Unless resetting all simulators, it's time to stop
+            return unless params[:device] == "all"
```

Comment:
Non-local exit from iterator, without return value. `next`, `break`, `Array#find`, `Array#any?`, etc. is preferred.